packages/flutter/test/material/elevated_button_theme_test.dart

```diff
@@ -179,4 +179,58 @@ void main() {
       checkButton(tester);
     });
   });
+
+  testWidgets('Theme shadowColor', (WidgetTester tester) async {
+    const ColorScheme colorScheme = ColorScheme.light();
+    const Color shadowColor = Color(0xff000001);
+
+    Widget buildFrame({ Color overallShadowColor, Color themeShadowColor, Color shadowColor }) {
+      return MaterialApp(
+        theme: ThemeData.from(colorScheme: colorScheme).copyWith(
+          shadowColor: overallShadowColor,
+        ),
+        home: Scaffold(
+          body: Center(
+            child: ElevatedButtonTheme(
+              data: ElevatedButtonThemeData(
+                style: ElevatedButton.styleFrom(
+                  shadowColor: themeShadowColor,
+                ),
+              ),
+              child: Builder(
+                builder: (BuildContext context) {
+                  return ElevatedButton(
+                    style: ElevatedButton.styleFrom(
+                      shadowColor: shadowColor,
+                    ),
+                    onPressed: () { },
+                    child: const Text('button'),
+                  );
+                },
+              ),
+            ),
+          ),
+        ),
+      );
+    }
+
+    final Finder buttonMaterialFinder = find.descendant(
+      of: find.byType(ElevatedButton),
+      matching: find.byType(Material),
+    );
+
+    await tester.pumpWidget(buildFrame(overallShadowColor: shadowColor));
```

Comment:
Minor nit, but it might be nice to make sure that each of the settings overrides the other (i.e. provide all three shadow colors to verify that the widget setting overrides the other two if they are all specified).

Reply:
OK, I've added some tests like that.